src/main.rs

```diff
@@ -13,32 +13,61 @@ use embassy_stm32::gpio::Level::{High,Low};
 
 use {defmt_rtt as _, panic_probe as _};
 
-use embassy_sync::signal::Signal;
+use embassy_sync::channel::Channel;
 use embassy_sync::blocking_mutex::raw::CriticalSectionRawMutex;
 use static_cell::StaticCell;
 use heapless::String;
 use core::sync::atomic::Ordering;
 use core::sync::atomic::AtomicU32;
 
+use time_stats::TimeStats;
+use stats::Stats;
+
+//
+// Debug configuration
+//
+#[allow(dead_code)]
+#[derive(PartialEq, Eq)]
+enum HrDebugMode {
+    Info, // Show difference bewteen adc and process_hr counts
+    Debug, // Show that adc, now.as_millis and process_hr are all in lockstep
+    Stats, // Show lots of stuff including min/mean/max time in hr task
+    DisplayOverrun, // Show display task overrun counter
+}
+#[allow(dead_code)]
+#[derive(PartialEq, Eq)]
+enum DebugMode {
+    None,
+    Hr(HrDebugMode),
+    DumpSamples, // Display raw samples
+    DumpTiming, // Display time spent in ADC and process_hr task
+}
+
+const DEBUG_MODE : DebugMode = DebugMode::Hr(HrDebugMode::DisplayOverrun);
+
 //
 // Things needed for 14-segment driver processing task
 //
 
 mod c5412;
 
+static C5412PINS_INST: StaticCell<c5412::C5412Pins> = StaticCell::new();
+
 // Async communication: value to display, 0-99, to c5412 task
 static DISP_VALUE_ATOMIC: AtomicU32 = AtomicU32::new(0);
 
-static C5412PINS_INST: StaticCell<c5412::C5412Pins> = StaticCell::new();
-
 //
 // Things needed for HR processing task
 //
 
 mod hr_alg3;
 
+// Async communication: ADC overrun detection.  Expect ADC_N == elapsed millis
+static ADC_N_ATOMIC: AtomicU32 = AtomicU32::new(0);
+
 // Async communication: ADC value from main (ADC) task to HR processing task
-static SAMPLE_SIGNAL: Signal<CriticalSectionRawMutex, u32> = Signal::new();
+static SAMPLE_CHANNEL: Channel<CriticalSectionRawMutex, u32, 200> =
```

Comment:
Comment and/or Const the 200 size, and explain choice and fact that it will be caught if it is too small and how

Reply:
Commented.  The size is otherwise not used